### PR TITLE
[DEVX-5243] Add last_request and last_response to http client.

### PR DESCRIPTION
**Pull request by jmctwilio**

### CODE REVIEW

File: src/main/java/com/twilio/http/HttpClient.java [hunk 1]
```diff
@@ -50,6 +53,10 @@ public Response reliableRequest(final Request request, final int[] retryCodes, i
             // Decrement retries
             retries--;
         }
+
+        last_request = request;
+        last_response = response;
```
Comment (dougblack) [inline src/main/java/com/twilio/http/HttpClient.java]: These should be `lastRequest` and `lastResponse` to follow Java naming conventions.

Comment (jingming) [inline src/main/java/com/twilio/http/HttpClient.java]: It might be more Java'y to make these members private and then expose getters.

Comment (efossier) [inline src/main/java/com/twilio/http/HttpClient.java]: Also I believe the other classes in this library encapsulate class fields generally so we would make these variables private/protected and add getters but not setters (currently someone could set it if they wanted to but not sure why they would)

File: src/main/java/com/twilio/http/HttpClient.java [hunk 2]
```diff
@@ -12,6 +12,9 @@
     public static final int RETRIES = 3;
     public static final long DELAY_MILLIS = 100L;
 
+    public Response lastResponse;
```
Comment (efossier) [inline src/main/java/com/twilio/http/HttpClient.java]: Should either be private (can only be accessed in the class), or protected (can only be access by class and its subclasses).

File: src/main/java/com/twilio/http/HttpClient.java [hunk 3]
```diff
@@ -50,9 +53,21 @@ public Response reliableRequest(final Request request, final int[] retryCodes, i
             // Decrement retries
             retries--;
         }
+
+        lastRequest = request;
```
Comment (efossier) [inline src/main/java/com/twilio/http/HttpClient.java]: lastRequest should probably be set before we make the request too.

**Pull request merged**